state/runtime/evm/instructions_test.go

```diff
@@ -10,81 +10,245 @@ import (
+func testArithmeticOperation(t *testing.T, f instruction, tests OperandsArithmetic) {
 	t.Helper()
 
 	s, closeFn := getState()
 	defer closeFn()
 
+	s.config = &chain.ForksInTime{Constantinople: true}
+
 	for _, i := range tests {
-		s.push(i.b)
-		s.push(i.a)
+		for _, operand := range i.operands {
+			s.push(operand)
+		}
 
 		f(s)
 
-		if i.expectedResult {
-			assert.Equal(t, uint64(1), s.pop().Uint64())
-		} else {
-			assert.Equal(t, uint64(0), s.pop().Uint64())
-		}
+		assert.EqualValues(t, i.expectedResult.Uint64(), s.pop().Uint64())
 	}
 }
 
 func TestAdd(t *testing.T) {
-	testArithmeticOperation(t, opAdd, twoOperandsArithmetic{
-		{one, one, two},
-		{zero, one, one},
+	testArithmeticOperation(t, opAdd, OperandsArithmetic{
+		{[]*big.Int{one, one}, two},
+		{[]*big.Int{zero, one}, one},
+		{[]*big.Int{three, two}, five},
+		{[]*big.Int{zero, zero}, zero},
 	})
 }
 
 func TestMul(t *testing.T) {
-	testArithmeticOperation(t, opMul, twoOperandsArithmetic{
-		{two, two, big.NewInt(4)},
-		{big.NewInt(3), two, big.NewInt(6)},
+	testArithmeticOperation(t, opMul, OperandsArithmetic{
+		{[]*big.Int{two, two}, four},
+		{[]*big.Int{three, two}, big.NewInt(6)},
+		{[]*big.Int{three, one}, three},
+		{[]*big.Int{zero, one}, zero},
 	})
 }
 
 func TestSub(t *testing.T) {
-	testArithmeticOperation(t, opSub, twoOperandsArithmetic{
-		{two, one, one},
-		{two, zero, two},
+	testArithmeticOperation(t, opSub, OperandsArithmetic{
+		{[]*big.Int{one, two}, one},
+		{[]*big.Int{zero, two}, two},
+		{[]*big.Int{two, two}, zero},
+	})
+}
+
+func TestDiv(t *testing.T) {
+	testArithmeticOperation(t, opDiv, OperandsArithmetic{
+		{[]*big.Int{two, two}, one},
+		{[]*big.Int{one, two}, two},
+		{[]*big.Int{one, zero}, zero},
+		{[]*big.Int{zero, one}, zero},
+	})
+}
+
+func TestSDiv(t *testing.T) {
+	testArithmeticOperation(t, opSDiv, OperandsArithmetic{
+		{[]*big.Int{two, two}, one},
+		{[]*big.Int{one, two}, two},
+		{[]*big.Int{zero, one}, zero},
+		{[]*big.Int{one, zero}, zero},
+	})
+}
+
+func TestMod(t *testing.T) {
+	testArithmeticOperation(t, opMod, OperandsArithmetic{
+		{[]*big.Int{two, three}, one},
+		{[]*big.Int{two, two}, zero},
+		{[]*big.Int{one, three}, zero},
+		{[]*big.Int{zero, one}, zero},
+		{[]*big.Int{three, five}, two},
+	})
+}
+
+func TestSMod(t *testing.T) {
+	testArithmeticOperation(t, opSMod, OperandsArithmetic{
+		{[]*big.Int{two, three}, one},
+		{[]*big.Int{two, two}, zero},
+		{[]*big.Int{one, three}, zero},
+		{[]*big.Int{zero, one}, zero},
+		{[]*big.Int{three, five}, two},
+	})
+}
+
+func TestReleaseAcquireBig(t *testing.T) {
+	releaseBig(five)
+
+	assert.Equal(t, five, acquireBig())
+}
+
+func TestExp(t *testing.T) {
+	t.Run("EIP158", func(t *testing.T) {
+		s, cancelFn := getState()
+		defer cancelFn()
+
+		s.config = &chain.ForksInTime{EIP158: true}
+		s.gas = 1000
+
+		s.push(one)
+		s.push(one)
+
+		opExp(s)
+
+		assert.Equal(t, uint64(950), s.gas)
+		assert.Equal(t, one, s.pop())
+	})
```

Comment:
Spice things up a bit, with some numbers higher than 1:

```suggestion
		s.push(three)
		s.push(two)

		opExp(s)

		assert.Equal(t, uint64(950), s.gas)
		assert.Equal(t, big.NewInt(8), s.pop())
```

Also you could have reused `testArithmeticOperation`, but provide the `state` as a parameter (if it is nil, then instantiate default one in the `testArithmeticOperation` as it is currently, otherwise use the provided reference).